### ISSUE TITLE
How are fields ordered in a table?

### ISSUE THREAD

**Issue by kramer65** [pardom-zz/ActiveAndroid#125]
We are currently looking at implementing AA in our app. So far everything looks really really good (props to the devs!), but what I wonder about is the way the tables are created. Since we like to know what's going on I had a look at my users table and it looks like this:

CREATE TABLE users (Id INTEGER PRIMARY KEY AUTOINCREMENT, lastName TEXT, lastModified INTEGER, avatarId TEXT, created INTEGER NOT NULL ON CONFLICT FAIL, uuid TEXT NOT NULL ON CONFLICT FAIL UNIQUE ON CONFLICT REPLACE, lastSync INTEGER, firstName TEXT, displayName TEXT);

Since the order of the table fields is not in the order I define them in my class nor in alphabetical order, I now wonder how the order of these fields is determined.

Does anybody know more about that?

**Comment by leonardoxh** [comment]
Try initialize the active android with debug set to true in your application class, this will get the creations scripts for you example: 

<code>ActiveAndroid.initialize(this, true)</code>

About the order I really don't no

**Comment by vsigler** [comment]
The columns are not sorted in any way. After reflection analysis, the field/column definitions are stored in a map. When generating the initial scripts, AA just takes this map and iterates over the set of keys (fields). So the order is whatever the hashmap returns. Does the order matter for some reason?

**Comment by kramer65** [comment]
@vsigler 
Thanks for the clear explanation! The order does not really matter at this moment. In the future however, we might want to bypass AA for some reason and in that case we like the things that AA creates to be predictable.
Would it be a big code change to have the table fields be created in alphabetical order?

**Comment by ecteodoro** [comment]
I suggest using a LinkedHashMap as it iterates in the same order entries (fields) are added to map.

**Comment by kramer65** [comment]
@ecteodoro 
Good idea. Would you have any idea as to where I can change this in the source code?

**Comment by ecteodoro** [comment]
Probably on line 39 of TableInfo class:

``` java
private Map<Field, String> mColumnNames = new HashMap<Field, String>();
```

https://github.com/pardom/ActiveAndroid/blob/master/src/com/activeandroid/TableInfo.java
